src/Internal/Workflow/Process/Process.php

```diff
@@ -175,22 +173,68 @@ function (\Throwable $e): void {
     }
 
     /**
-     * @param MethodHandler|\Closure(ValuesInterface): mixed $handler
+     * Initialize workflow instance and start execution.
      */
-    public function start(MethodHandler|\Closure $handler, ValuesInterface $values, bool $deferred): void
-    {
-        try {
-            $this->makeCurrent();
+    public function initAndStart(
+        WorkflowContext $context,
+        WorkflowInstance $instance,
+        bool $deferred,
+    ): void {
+        $handler = $instance->getHandler();
+        $instance = $context->getWorkflowInstance();
+        $arguments = null;
 
-            // Prepare typed input
-            \assert($handler instanceof MethodHandler);
-            $arguments = $handler->resolveArguments($values);
+        try {
+            // Initialize workflow instance
+            //
+            // Resolve arguments if #[WorkflowInit] is used
```

Comment:
Something we do in the other SDKs is we actually run the init method in a special "readonly" context that will throw errors if the user attempts to issue any commands during init. I dunno if that's something that you can readily enforce, but if possible it's a nice bit of safety.

Reply:
Now any request from the constructor will throw an exception.
![image](https://github.com/user-attachments/assets/d8366d86-8a4f-4713-922f-7691770a63ab)

Workflow Context will no longer be readonly right after the init stage before the interceptor method `WorkflowInboundCallsInterceptor::execute` is run.